diff -ru src/c-blosc-orig/blosc/blosc.c src/c-blosc-1.21.6/blosc/blosc.c
--- src/c-blosc-orig/blosc/blosc.c	2024-06-24 15:56:17.000000000 +0000
+++ src/c-blosc-1.21.6/blosc/blosc.c	2026-03-17 16:48:22.323847400 +0000
@@ -56,13 +56,7 @@
   #include <inttypes.h>
 #endif  /* _WIN32 */
 
-/* Include the win32/pthread.h library for all the Windows builds. See #224. */
-#if defined(_WIN32)
-  #include "win32/pthread.h"
-  #include "win32/pthread.c"
-#else
-  #include <pthread.h>
-#endif
+#include <pthread.h>
 
 
 /* Some useful units */
diff -ru src/c-blosc-orig/blosc/shuffle.c src/c-blosc-1.21.6/blosc/shuffle.c
--- src/c-blosc-orig/blosc/shuffle.c	2024-06-24 15:56:17.000000000 +0000
+++ src/c-blosc-1.21.6/blosc/shuffle.c	2026-03-17 16:48:32.699282200 +0000
@@ -14,11 +14,7 @@
 #include "blosc-comp-features.h"
 #include <stdio.h>
 
-#if defined(_WIN32)
-#include "win32/pthread.h"
-#else
 #include <pthread.h>
-#endif
 
 /* Visual Studio < 2013 does not have stdbool.h so here it is a replacement: */
 #if defined __STDC__ && defined __STDC_VERSION__ && __STDC_VERSION__ >= 199901L
