diff -ru src/bitshuffle-orig/src/bitshuffle_core.c src/bitshuffle-0.5.2/src/bitshuffle_core.c
--- src/bitshuffle-orig/src/bitshuffle_core.c	2024-09-26 17:46:18.000000000 +0000
+++ src/bitshuffle-0.5.2/src/bitshuffle_core.c	2026-03-12 22:17:41.896766700 +0000
@@ -625,7 +625,6 @@
     size_t ii, kk;
     const char* in_b = (const char*) in;
     char* out_b = (char*) out;
-    uint16_t* out_ui16;
 
     int64_t count;
 
@@ -641,8 +640,8 @@
         for (kk = 0; kk < 8; kk++) {
             bt = move_byte_mask_neon((uint8x16_t) xmm);
             xmm = vshlq_n_s16(xmm, 1);
-            out_ui16 = (uint16_t*) &out_b[((7 - kk) * nbyte + ii) / 8];
-            *out_ui16 = bt;
+            uint16_t bt16 = (uint16_t)bt;
+            memcpy(&out_b[((7 - kk) * nbyte + ii) / 8], &bt16, 2);
         }
     }
     count = bshuf_trans_bit_byte_remainder(in, out, size, elem_size,
@@ -780,7 +779,7 @@
     // With a bit of care, this could be written such that such that it is
     // in_buf = out_buf safe.
     const char* in_b = (const char*) in;
-    uint16_t* out_ui16 = (uint16_t*) out;
+    char* out_b = (char*) out;
 
     size_t ii, jj, kk;
     size_t nbyte = elem_size * size;
@@ -799,7 +798,8 @@
                     bt = move_byte_mask_neon((uint8x16_t) xmm);
                     xmm = vshlq_n_s16(xmm, 1);
                     size_t ind = (ii + jj / 8 + (7 - kk) * elem_size);
-                    out_ui16[ind / 2] = bt;
+                    uint16_t bt16 = (uint16_t)bt;
+                    memcpy(&out_b[ind], &bt16, 2);
                 }
             }
         }
@@ -1114,7 +1114,6 @@
     size_t ii, kk;
     const char* in_b = (const char*) in;
     char* out_b = (char*) out;
-    uint16_t* out_ui16;
 
     int64_t count;
 
@@ -1130,8 +1129,8 @@
         for (kk = 0; kk < 8; kk++) {
             bt = _mm_movemask_epi8(xmm);
             xmm = _mm_slli_epi16(xmm, 1);
-            out_ui16 = (uint16_t*) &out_b[((7 - kk) * nbyte + ii) / 8];
-            *out_ui16 = bt;
+            uint16_t bt16 = (uint16_t)bt;
+            memcpy(&out_b[((7 - kk) * nbyte + ii) / 8], &bt16, 2);
         }
     }
     count = bshuf_trans_bit_byte_remainder(in, out, size, elem_size,
@@ -1279,7 +1278,7 @@
     // With a bit of care, this could be written such that such that it is
     // in_buf = out_buf safe.
     const char* in_b = (const char*) in;
-    uint16_t* out_ui16 = (uint16_t*) out;
+    char* out_b = (char*) out;
 
     size_t ii, jj, kk;
     size_t nbyte = elem_size * size;
@@ -1298,7 +1297,8 @@
                     bt = _mm_movemask_epi8(xmm);
                     xmm = _mm_slli_epi16(xmm, 1);
                     size_t ind = (ii + jj / 8 + (7 - kk) * elem_size);
-                    out_ui16[ind / 2] = bt;
+                    uint16_t bt16 = (uint16_t)bt;
+                    memcpy(&out_b[ind], &bt16, 2);
                 }
             }
         }
@@ -1403,7 +1403,6 @@
     size_t ii, kk;
     const char* in_b = (const char*) in;
     char* out_b = (char*) out;
-    int32_t* out_i32;
 
     size_t nbyte = elem_size * size;
 
@@ -1417,8 +1416,7 @@
         for (kk = 0; kk < 8; kk++) {
             bt = _mm256_movemask_epi8(ymm);
             ymm = _mm256_slli_epi16(ymm, 1);
-            out_i32 = (int32_t*) &out_b[((7 - kk) * nbyte + ii) / 8];
-            *out_i32 = bt;
+            memcpy(&out_b[((7 - kk) * nbyte + ii) / 8], &bt, 4);
         }
     }
     count = bshuf_trans_bit_byte_remainder(in, out, size, elem_size,
@@ -1574,7 +1572,7 @@
                     bt = _mm256_movemask_epi8(ymm);
                     ymm = _mm256_slli_epi16(ymm, 1);
                     size_t ind = (ii + jj / 8 + (7 - kk) * elem_size);
-                    * (int32_t *) &out_b[ind] = bt;
+                    memcpy(&out_b[ind], &bt, 4);
                 }
             }
         }
@@ -1648,7 +1646,6 @@
     size_t nbyte = elem_size * size;
     int64_t count;
 
-    int64_t* out_i64;
     __m512i zmm;
     __mmask64 bt;
     if (nbyte >= 64) {
@@ -1659,23 +1656,21 @@
             for (kk = 0; kk < 8; kk++) {
                 bt = _mm512_cmp_epi8_mask(zmm, mask, 1);
                 zmm = _mm512_slli_epi16(zmm, 1);
-                out_i64 = (int64_t*) &out_b[((7 - kk) * nbyte + ii) / 8];
-                *out_i64 = (int64_t)bt;
+                int64_t bt64 = (int64_t)bt;
+                memcpy(&out_b[((7 - kk) * nbyte + ii) / 8], &bt64, 8);
             }
         }
     }
 
     __m256i ymm;
     int32_t bt32;
-    int32_t* out_i32;
     size_t start = nbyte - nbyte % 64;
     for (ii = start; ii + 31 < nbyte; ii += 32) {
         ymm = _mm256_loadu_si256((__m256i *) &in_b[ii]);
         for (kk = 0; kk < 8; kk++) {
             bt32 = _mm256_movemask_epi8(ymm);
             ymm = _mm256_slli_epi16(ymm, 1);
-            out_i32 = (int32_t*) &out_b[((7 - kk) * nbyte + ii) / 8];
-            *out_i32 = bt32;
+            memcpy(&out_b[((7 - kk) * nbyte + ii) / 8], &bt32, 4);
         }
     }
 
@@ -1739,7 +1734,8 @@
                     bt = _mm512_cmp_epi8_mask(zmm, mask, 1);
                     zmm = _mm512_slli_epi16(zmm, 1);
                     size_t ind = (ii + jj / 8 + (7 - kk) * elem_size);
-                    * (int64_t *) &out_b[ind] = bt;
+                    int64_t bt64 = (int64_t)bt;
+                    memcpy(&out_b[ind], &bt64, 8);
                 }
             }
         }
