diff -ru src/hdf5-orig/src/hdf5.h src/hdf5-2.1.1/src/hdf5.h
--- src/hdf5-orig/src/hdf5.h	2025-11-11 00:00:49.000000000 +0000
+++ src/hdf5-2.1.1/src/hdf5.h	2025-11-21 21:13:48.761485100 +0000
@@ -56,23 +56,17 @@
 
 /* Predefined file drivers */
 #include "H5FDcore.h"     /* Files stored entirely in memory          */
-#include "H5FDdirect.h"   /* Linux direct I/O                         */
 #include "H5FDfamily.h"   /* File families                            */
-#include "H5FDhdfs.h"     /* Hadoop HDFS                              */
 #include "H5FDlog.h"      /* sec2 driver with I/O logging (for debugging) */
 #include "H5FDmirror.h"   /* Mirror VFD and IPC definitions           */
-#include "H5FDmpi.h"      /* MPI-based file drivers                   */
 #include "H5FDmulti.h"    /* Usage-partitioned file family            */
 #include "H5FDonion.h"    /* Onion file I/O                           */
-#include "H5FDros3.h"     /* R/O S3 "file" I/O                        */
 #include "H5FDsec2.h"     /* POSIX unbuffered file I/O                */
 #include "H5FDsplitter.h" /* Twin-channel (R/W & R/O) I/O passthrough */
 #include "H5FDstdio.h"    /* Standard C buffered I/O                  */
 #ifdef H5_HAVE_WINDOWS
 #include "H5FDwindows.h" /* Win32 I/O                                */
 #endif
-#include "H5FDsubfiling.h" /* Subfiling VFD                            */
-#include "H5FDioc.h"       /* I/O Concentrator VFD                     */
 
 /* Virtual object layer (VOL) connectors */
 #include "H5VLnative.h"   /* Native VOL connector                     */
