diff -ru src/c-blosc2-orig/blosc/blosclz.c src/c-blosc2-2.23.1/blosc/blosclz.c
--- src/c-blosc2-orig/blosc/blosclz.c	2026-03-03 09:50:48.000000000 +0000
+++ src/c-blosc2-2.23.1/blosc/blosclz.c	2026-03-12 16:11:42.373465100 +0000
@@ -46,13 +46,8 @@
 #define MAX_DISTANCE 8191
 #define MAX_FARDISTANCE (65535 + MAX_DISTANCE - 1)
 
-#ifdef BLOSC_STRICT_ALIGN
-  #define BLOSCLZ_READU16(p) ((p)[0] | (p)[1]<<8)
-  #define BLOSCLZ_READU32(p) ((p)[0] | (p)[1]<<8 | (p)[2]<<16 | (p)[3]<<24)
-#else
-  #define BLOSCLZ_READU16(p) *((const uint16_t*)(p))
-  #define BLOSCLZ_READU32(p) *((const uint32_t*)(p))
-#endif
+#define BLOSCLZ_READU16(p) ((uint16_t)(p)[0] | ((uint16_t)(p)[1]<<8))
+#define BLOSCLZ_READU32(p) ((uint32_t)(p)[0] | ((uint32_t)(p)[1]<<8) | ((uint32_t)(p)[2]<<16) | ((uint32_t)(p)[3]<<24))
 
 #define HASH_LOG (14U)
 
