diff -ru src/c-blosc2-orig/blosc/directories.c src/c-blosc2-2.23.1/blosc/directories.c
--- src/c-blosc2-orig/blosc/directories.c	2026-03-03 09:50:48.000000000 +0000
+++ src/c-blosc2-2.23.1/blosc/directories.c	2026-03-17 18:18:01.957056300 +0000
@@ -26,7 +26,7 @@
   int blosc2_remove_dir(const char* dir_path) {
     char* path;
     char last_char = dir_path[strlen(dir_path) - 1];
-    if (last_char != '\\' || last_char != '/') {
+    if (last_char != '\\' && last_char != '/') {
       path = malloc(strlen(dir_path) + 2 + 1);
       sprintf(path, "%s\\*", dir_path);
     }
@@ -81,7 +81,7 @@
 char* blosc2_normalize_dirpath(const char* dir_path) {
   char last_char = dir_path[strlen(dir_path) - 1];
   char* path;
-  if (last_char != '\\' || last_char != '/'){
+  if (last_char != '\\' && last_char != '/'){
     path = malloc(strlen(dir_path) + 1 + 1);
     sprintf(path, "%s/", dir_path);
   }
